mate-panel/panel-run-dialog.c

```diff
@@ -341,6 +342,18 @@ command_is_executable (const char   *command,
 	if (!result)
 		return FALSE;
 
+	for (int i = 0; i < g_strv_length (argv); i++)
```

Comment:
This could be optimized as `for (gsize i = 0; argv[i] != NULL; i++)`.  `g_strv_length()` is a somewhat costly operation (it traverses the list to count items, [basically `gsize len = 0; while (argv[len]) len++; return len;`)](https://gitlab.gnome.org/GNOME/glib/-/blob/main/glib/gstrfuncs.c#L2949), and this calls it once per loop iteration.  As the length of `argv` doesn't change in the loop body, there's not need to re-compute it each time.
Alternatively, you could cache the length in a variable: `gsize argv_len = g_strv_length (argv); for (gsize i = 0; i < argv_len; i++)`.